### ISSUE TITLE
Object Initializer Completion

### ISSUE THREAD

**Issue by raffaeler** [dotnet/roslyn#16229]
**Version Used**: 1.3.2
**Steps to Reproduce**:
1. I create a CompletionService and a CodeAnalisys.Document with the following source:
```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using __entityGenerator1;

namespace __functionGenerator7
{
    public static partial class __generated7
    {
        public static __entityGenerator1.Ent02 navAA(System.DateTimeOffset t)
        {
            return new Ent02()
{
  @@Caret here@@
};
        }
    }
}
```

2. Invoke GetCompletionsAsync passing the given CaretPosition

**Expected Behavior**:
I expect the CompletionList.Items to contain the Ent02 properties.
The target is to show the correct suggestions to initialize Ent02

**Actual Behavior**:
The CompletionList does not contain any of the Ent02 properties (it has several properties)

I could see discussions about this in the Omnisharp repo with the same request.
Should it already work in 1.3.2? Later versions?
I am still waiting the next stable version before going there.

Thanks!

**Comment by Pilchie** [comment]
@mattwar - can you take a look here?

**Comment by mattwar** [comment]
The following test code succeeds finding the members of the class to complete.

Is it possible that the type Ent02 is not in scope in your test or has members defined in an imported base class from an assembly that is not referenced?

```cs
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Completion;
using Microsoft.CodeAnalysis.Text;

namespace Issue16229
{
    class Program
    {
        static void Main(string[] args)
        {
            var text = @"
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class C { public int A; public int B }

public static class Test
{
    public static void Method()
    {
        return new C()
        {
            $$
        };
    }
}";

            var list = GetCompletions(text);
        }

        static readonly string CaretText = "$$";

        static CompletionList GetCompletions(string text)
        {
            var position = text.IndexOf(CaretText);
            text = text.Remove(position, CaretText.Length);

            var workspace = new AdhocWorkspace();
            var project = workspace.AddProject("test", LanguageNames.CSharp);
            var document = workspace.AddDocument(project.Id, "test.cs", SourceText.From(text));

            var service = CompletionService.GetService(document);

            return service.GetCompletionsAsync(document, position).Result;
        }
    }
}
```

**Comment by mattwar** [comment]
Here's the same test but actually using the 1.3.2 bits (not the current bits).

```cs
using System;
using System.Reflection;
using System.Linq;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Completion;
using Microsoft.CodeAnalysis.Host.Mef;
using Microsoft.CodeAnalysis.Text;
using System.IO;

namespace Issue16229
{
    class Program
    {
        static void Main(string[] args)
        {
            var text = @"
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namepace N
{
    public class C { public int A; public int B }
}

namespace Y
{
    using N;

    public static class Test
    {
        public static C Method()
        {
            return new C()
            {
                $$
            };
        }
    }
}";

            var list = GetCompletions(text);
        }

        static readonly string CaretText = "$$";

        static CompletionList GetCompletions(string text)
        {
            var position = text.IndexOf(CaretText);
            text = text.Remove(position, CaretText.Length);

            var workspace = new AdhocWorkspace(HostServices);
            var project = workspace.AddProject("test", LanguageNames.CSharp);
            var document = workspace.AddDocument(project.Id, "test.cs", SourceText.From(text));

            var service = CompletionService.GetService(document);

            return service.GetCompletionsAsync(document, position).Result;
        }

        private static readonly MefHostServices HostServices = 
            MefHostServices.Create(
                MefHostServices.DefaultAssemblies.Concat(new[] {
                    Assembly.LoadFrom(Path.Combine(Environment.CurrentDirectory, "Microsoft.CodeAnalysis.Features.dll")),
                    Assembly.LoadFrom(Path.Combine(Environment.CurrentDirectory, "Microsoft.CodeAnalysis.CSharp.Features.dll"))
                    }));
    }
}
```

The difference is the need to create the host services that references the feature dlls, which is not in the default assemblies in 1.3.2.

**Comment by raffaeler** [comment]
I found the problem, it was my fault of course.
Your confirmation that this must work in 1.3.2 made me find the error. The issues that I found on the Omnisharp repo, deceived me.

I was adding the references but I forgot to re-assign the project instance (since the project instance is immutable, it was tricky to find).
What cheated me is that a very similar method that I use to compile the same snippet, worked without errors... but similar is not identical :)

BTW thanks for highlighting the difference with the 2.0. It will be useful as soon as I migrate the codebase when it will be rtm'd.
